src/hotspot/share/runtime/mutex.cpp

```diff
@@ -412,7 +412,12 @@ void Mutex::check_rank(Thread* thread) {
     // to acquire, then deadlock prevention rules require that the rank
     // of m2 be less than the rank of m1. This prevents circular waits.
     if (least != NULL && least->rank() <= this->rank()) {
-      thread->print_owned_locks();
+      if (least->rank() > Mutex::tty) {
+        // Printing owned locks acquires tty lock. If lowest rank is below or equal
```

Comment:
If the least* rank was* below or equal tty,

Reply:
Fixed.